### PR TITLE
Adapt to external URI that might not be a URL

### PR DESCRIPTION
**Pull request by JohnNiang**
The external URI might not be an URL, so it would not work well. See https://github.com/halo-dev/halo/issues/3762 for more.

This PR 
- adapts to external URI that might not be a URL
- refactors cache key with `SitemapGeneratorOptions` to make sure the different request URLs have different keys
- upgrades to Gradle 8.0.2
- refactors build script

BTW, I have tested this plugin on Halo 2.0-2.4.

You could test this plugin manually using [plugin-sitemap.zip](https://github.com/halo-sigs/plugin-sitemap/files/11248612/plugin-sitemap.zip).

/kind bug

```release-note
修复 External URL 为“/”时无法访问的问题
```

### CODE REVIEW

File: src/test/java/run/halo/sitemap/CachedSitemapGetterTest.java [hunk 1]
```diff
@@ -20,22 +25,30 @@ public class CachedSitemapGetterTest {
...
     void setUp() {
-        when(lister.list()).thenReturn(
+        when(lister.list(any())).thenReturn(
             Flux.just(SitemapEntry.builder().loc("http://localhost:8090/about").build()));
         getter = new CachedSitemapGetter(lister);
     }
 
     @Test
-    void get() throws InterruptedException {
-        CountDownLatch countDownLatch = new CountDownLatch(10);
```
[the remark this answers is not in the document]
Comment (JohnNiang) [reply src/test/java/run/halo/sitemap/CachedSitemapGetterTest.java:30]: 并没有去掉，而是用 ExecutorService 替代了。

**Review by guqing — COMMENTED** [review]
/lgtm

**Review by ruibaby — APPROVED** [review]
/approve

**Comment by JohnNiang** [comment]
I'm going to merge this PR manually, because the automatic merge is unavailable currently.

**Pull request merged**